common/lib/dependabot/pull_request_creator/message_builder/link_and_mention_sanitizer.rb

```diff
@@ -118,20 +118,28 @@ def build_mention_nodes(text)
         end
 
         def build_mention_link_text_nodes(text)
-          code_node = CommonMarker::Node.new(:code)
-          code_node.string_content = text
-          [code_node]
+          text_node = CommonMarker::Node.new(:text)
+          text_node.string_content = insert_zero_width_space_in_mention(text)
+          [text_node]
         end
 
         def create_link_node(url, text)
           link_node = CommonMarker::Node.new(:link)
-          code_node = CommonMarker::Node.new(:code)
+          text_node = CommonMarker::Node.new(:text)
           link_node.url = url
-          code_node.string_content = text
-          link_node.append_child(code_node)
+          text_node.string_content = insert_zero_width_space_in_mention(text)
+          link_node.append_child(text_node)
           link_node
         end
 
+        # NOTE: Add a zero-width space between the @ and the username to prevent
+        # email replies on dependabot pull requests triggering notifications to
+        # users who've been mentioned in changelogs etc. PR email replies parse
+        # the content of the pull request body in plain text.
+        def insert_zero_width_space_in_mention(mention)
+          mention.sub("@", "@\u200B").encode("utf-8")
```

Comment:
Attempted this to make it more obvious we're inserting this zero-width space and make it somewhat greppable, would like to confirm this encoding won't cause any issues when creating the pr.